### PR TITLE
Avoid running logical line rule logic if not enabled

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR updates the logical line rules entry-point function to only run the logic if any of the rules within that group is enabled.

Although this shouldn't really give any performance improvements, it's better not to do additional work if we can. This is also consistent with how other rules are run.

## Test Plan

`cargo insta test`

### CODE REVIEW

File: crates/ruff_linter/src/checkers/logical_lines.rs [hunk 1]
```diff
@@ -47,34 +47,90 @@ pub(crate) fn check_logical_lines(
...
+                Rule::TabBeforeOperator,
+                Rule::TabAfterOperator,
+            ]) {
+                space_around_operator(&line, &mut context);
+            }
+
+            if settings.rules.any_enabled(&[
+                Rule::UnexpectedSpacesAroundKeywordParameterEquals,
+                Rule::MissingWhitespaceAroundParameterEquals,
+            ]) {
```
Comment (MichaReiser) [inline crates/ruff_linter/src/checkers/logical_lines.rs:62]: Let's hope the compiler is smart enough to move the check out of the `for` loop :)

**Review by MichaReiser — APPROVED** [review]
Thanks for doing this. This might improve performance, it just won't show in our benchmarks because we either run all or no logical line rules.

File: crates/ruff_linter/src/checkers/logical_lines.rs (hunk 1 above)
Comment (dhruvmanila) [reply crates/ruff_linter/src/checkers/logical_lines.rs:62]: Yeah, I hope so as well. Do you know how can one get that kind of information?

Comment (dhruvmanila) [reply crates/ruff_linter/src/checkers/logical_lines.rs:62]: I've decided to collect them out in variables outside the loop similar to `physical_lines.rs`.

**Pull request merged**